docs/spec.md

```diff
@@ -704,195 +708,375 @@ expressions must not be preceded by whitespace before the `[` character,
+of literal values, operators, variables, and further nested expressions
+that evaluates to a single value.
 
-    operands   operators      result
-    num        + - * / %      num
-    string     +              string
-    array      +              array
-    bool       and or         bool
-    num        <  <=  >  >=   bool
-    string     <  <=  >  >=   bool
+In Evy, there are two types of operators: unary operators and binary
+operators:
 
-`==` and `!=` compare two operands of the same type for equality and
-have a `bool` result.
+- _Unary operators_ act on a single operand. For example, the unary operator `-` negates the value of its operand.
+- _Binary operators_ act on two operands. For example, the binary operator `+` adds the two operands together.
 
-`+` `-` `*` `/` `%` stand for addition, subtraction, multiplication,
-division and the [modulo operator]. `+` may also be used as
-concatenation operator for `string` and `array` types.
+Operators can be combined to form larger expressions, for example, the
+expression `-delta + 3` would first negate the value of the variable `delta` and
+then add literal number `3` to the result.
 
-Boolean operators `and`, `or` stand for [logical conjunction (AND)] and
-[logical disjunction (OR)]. They perform [short-circuit evaluation]
-where the right-hand side of the operator is not evaluated if the result
-of the operation can be determined from the left-hand side alone.
-Comparison operators `<` `<=` `>` `>=` stand for less, less or equal,
+Binary expressions can only be evaluated if the operands are of the
+same type. For example, you cannot add a string to a number. There is
+no automated type conversion of operands.
+
+There are a variety of binary operators: arithmetic, concatenation,
+logical, and comparison operators.
+
+| Operator            | Operands  | Result   | Description   |
+| ------------------- | --------- | -------- | ------------- |
+| `+` `-` `*` `/` `%` | `num`     | `num`    | arithmetic    |
+| `+`                 | `string`  | `string` | concatenation |
+| `+`                 | array     | array    | concatenation |
+| `and` `or`          | `bool`    | `bool`   | logical       |
+| `<` `<=` `>` `>=`   | `num`     | `bool`   | comparison    |
+| `<` `<=` `>` `>=`   | `string`  | `bool`   | comparison    |
+| `==` `!=`           | all types | `bool`   | comparison    |
+
+### Arithmetic and Concatenation Operators
+
+The _arithmetic operators_ `+`, `-`, `*`, `/`, and `%` stand for addition,
+subtraction, multiplication, division, and the [modulo operator]. The
+symbol `+` can also be used to concatenate strings and arrays.
+
+The _modulo operator_ `%`, also known as the remainder operator, returns the
+remainder of a division operation. For example, `10 % 3` results in
+`1`, because `10` divided by `3` has a remainder of `1`.
+
+The _concatenation operator_ `+`, combines two strings or two arrays
+together. For example, `"fire"` + `"engine"` combines into the string
```

Comment:
:pineapple: Hey! I would have used exactly the same example strings!

Reply:
🚒